src/main/java/de/tum/in/www1/artemis/web/rest/ExerciseResource.java

```diff
@@ -312,29 +315,54 @@ private StatsForInstructorDashboardDTO populateCommonStatistics(Exercise exercis
      *
      * @param exerciseId the id of the exercise to get the repos from
      * @param studentIds the studentIds seperated via semicolon to get their submissions
+     * @param allStudents whether the repositories of all students should be downloaded
+     * @param filterLateSubmissions whether late submissions should be filtered out
+     * @param filterLateSubmissionsDate the date after which all submissions should be filtered out (may be null)
+     * @param addStudentName whether the student name should be added to the project
+     * @param squashAfterInstructor whether all changes after the setup from the instructors should be squashed into one commit
+     * @param normalizeCodeStyle whether the code style should be normalized (line endings, encoding)
      * @return ResponseEntity with status
      * @throws IOException if submissions can't be zipped
      */
     @GetMapping(value = "/exercises/{exerciseId}/participations/{studentIds}")
     @PreAuthorize("hasAnyRole('TA', 'INSTRUCTOR', 'ADMIN')")
-    public ResponseEntity<Resource> exportSubmissions(@PathVariable Long exerciseId, @PathVariable String studentIds) throws IOException {
-        studentIds = studentIds.replaceAll(" ", "");
+    public ResponseEntity<Resource> exportSubmissions(@PathVariable Long exerciseId, @PathVariable String studentIds, @RequestParam boolean allStudents,
+            @RequestParam boolean filterLateSubmissions, @RequestParam @DateTimeFormat(pattern = "yyyy-MM-dd'T'HH:mm:ss.SSSZ") Date filterLateSubmissionsDate,
+            @RequestParam boolean addStudentName, @RequestParam boolean squashAfterInstructor, @RequestParam boolean normalizeCodeStyle) throws IOException {
         Exercise exercise = exerciseService.findOne(exerciseId);
 
         // TODO: allow multiple options:
         // - one boolean flag per stager task (see exportParticipations)
         // - one boolean flag that all student submissions should be downloaded
 
+        ZonedDateTime filterLateSubmissionsZonedDateTime;
+
+        if (filterLateSubmissionsDate == null) {
+            filterLateSubmissionsZonedDateTime = exercise.getDueDate();
+        }
+        else {
+            filterLateSubmissionsZonedDateTime = ZonedDateTime.ofInstant(filterLateSubmissionsDate.toInstant(), ZoneId.systemDefault());
```

Comment:
Isn't this just `ZonedDateTime.now()`?

Reply:
I don't think so:
`filterLateSubmissionsDate` is the date the user selected in the client and as it is not null, the user actually selected a specific date.
As this can (and most likely will) differ from the current date, I believe it's correct the way it is.